src/app/reporting/ReportScheduler.h

```diff
@@ -132,21 +164,16 @@ class ReportScheduler : public ReadHandler::Observer
 
     /// @brief Check whether a ReadHandler is reportable right now, taking into account its minimum and maximum intervals.
     /// @param aReadHandler read handler to check
-    bool IsReportableNow(ReadHandler * aReadHandler)
-    {
-        return FindReadHandlerNode(aReadHandler)->IsReportableNow();
-    } // TODO: Change the IsReportableNow to IsReportable() for readHandlers
+    bool IsReportableNow(ReadHandler * aReadHandler) { return FindReadHandlerNode(aReadHandler)->IsReportableNow(); }
     /// @brief Check if a ReadHandler is reportable without considering the timing
-    bool IsReadHandlerReportable(ReadHandler * aReadHandler) const
-    {
-        return aReadHandler->IsGeneratingReports() && aReadHandler->IsDirty();
-    }
+    bool IsReadHandlerReportable(ReadHandler * aReadHandler) const { return aReadHandler->IsReportable(); }
 
     /// @brief Get the number of ReadHandlers registered in the scheduler's node pool
     size_t GetNumReadHandlers() const { return mNodesPool.Allocated(); }
 
 protected:
     friend class chip::app::reporting::TestReportScheduler;
+    friend class chip::app::TestReadInteraction;
```

Comment:
Why friend? This is not needed.

Reply:
So the TestReadInteraction can access the protected method FindReadHandlerNode().

Alternative would be to create public test function:
SetFlagsForHandler(const ReadHandler * aReadHandler)
CheckFlagsForHandler(const ReadHandler * aReadHandler)